jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -927,16 +931,166 @@ static String formatExpires(Instant expires)
-     * (in RFC 1123 format) into an {@link Instant}.</p>
+     * <p>Parses the {@code Expires} attribute value using algorithm
+     * specified in <a href="https://datatracker.ietf.org/doc/html/rfc6265#section-5.1.1">RFC6265: Section 5.1.1: Date</a>
+     * in into an {@link Instant}.</p>
      *
-     * @param expires an instant in the RFC 1123 string format
+     * @param expires an instant in the string format supported by RFC6265 HTTP State Management
      * @return an {@link Instant} parsed from the given string
      */
     static Instant parseExpires(String expires)
     {
-        // TODO: RFC 1123 format only for now, see https://www.rfc-editor.org/rfc/rfc2616#section-3.3.1.
-        return ZonedDateTime.parse(expires, DateTimeFormatter.RFC_1123_DATE_TIME).toInstant();
+        Objects.requireNonNull(expires, "Expires string cannot be null");
+
+        int year = -1;
+        int month = -1;
+        int day = -1;
+        int hour = -1;
+        int minute = -1;
+        int second = -1;
+
+        // delimiter       = %x09 / %x20-2F / %x3B-40 / %x5B-60 / %x7B-7E
+        // Note: That leaves "0-9" + "A-Z" + "a-z" + ":"
+        QuotedStringTokenizer dateTokenizer = QuotedStringTokenizer.builder()
+            .delimiters("\t" + // %x09
+                " !#$%&'()*+,-./" + " + " + // %x20-2F
+                ";<=>?@" + // %x3B-40
+                "[\\]^_`" + // %x5B-60
+                "{|}~" // %x7B-7E
+            )
+            .build();
+
+        Iterator<String> dateIter = dateTokenizer.tokenize(expires);
+        while (dateIter.hasNext())
+        {
+            String part = dateIter.next();
+
+            // RFC 6265 - Section 5.1.1 - Step 2.1 - time (00:00:00)
+            if (hour == (-1) && part.length() == 8 && part.charAt(2) == ':' && part.charAt(5) == ':')
+            {
+                try
+                {
+                    hour = StringUtil.toInt(part, 0);
+                    minute = StringUtil.toInt(part, 3);
+                    second = StringUtil.toInt(part, 6);
+                }
+                catch (IllegalArgumentException e)
+                {
+                    // bad time syntax
+                    throw new DateTimeSyntaxException("Invalid [time]: " + expires);
+                }
+                continue;
+            }
+
+            // RFC 6265 - Section 5.1.1 - Step 2.2
+            if (day == (-1) && part.length() <= 2)
+            {
+                try
+                {
+                    day = StringUtil.toInt(part, 0);
+                }
+                catch (IllegalArgumentException e)
+                {
+                    throw new DateTimeSyntaxException("Invalid [day]: " + expires);
+                }
+                continue;
+            }
+
+            // RFC 6265 - Section 5.1.1 - Step 2.3
+            if (month == (-1) && part.length() == 3)
+            {
+                int m = switch (part.toLowerCase(Locale.ENGLISH))
```

Comment:
Frequently dates will be sent as "Jan" and sometimes as "JAN", so we should avoid the creation of an extra string hear.  Use a case insensitive `Index` instead.

Reply:
Interestingly the HTTP spec says the Date is case sensitive.

https://datatracker.ietf.org/doc/html/rfc9110#name-date-time-formats

> HTTP-date is case sensitive. Note that [Section 4.2](https://www.rfc-editor.org/rfc/rfc9111#section-4.2) of [[CACHING](https://datatracker.ietf.org/doc/html/rfc9111)] relaxes this for cache recipients.